[Chloe Lee Ke Er] iP

chloelee767 opened the pull request.

Commit:
Add Gradle support

chloelee767 committed:
Ignore .class files

chloelee767 committed:
Complete level 1

chloelee767 committed:
Complete level 2

Includes a lot of setup for future levels

chloelee767 committed:
Add Task class

chloelee767 committed:
Complete level 3

chloelee767 committed:
Add Deadline and Event classes

chloelee767 committed:
Complete level 4

chloelee767 committed:
Add automated test UI testing

chloelee767 committed:
Add error handling

chloelee767 committed:
Fix typo and update test cases

chloelee767 committed:
Implement delete and update test cases

chloelee767 committed:
Fix bug in deadline and event creation

The bug: deadline something/bysometime would have been treated as
valid deadline (likewise for event).

chloelee767 committed:
Minor refactor

Change function name to be more descriptive

chloelee767 committed:
Duke class: refactor route function for less repetition

There is a lot of repeated logic for parsing and validating input in
the route function.

Let's:
- extract the common logic into helper functions
- create a new type of exception (DukeRoutingException)

Exceptions are thrown by the helper functions to make it easy to break
out of the control flow at many different points within the route function.

The exceptions are caught within the route function, rather than
becoming a checked exception for the route function, so as to keep the
run function clean.

chloelee767 committed:
Refactor existing files into packages

chloelee767 committed:
Refactor say functions from Duke class to Ui class

chloelee767 committed:
Refactor route function from Duke class to Parser class

chloelee767 committed:
Pass display text around as String instead of List<String>

This allows us to make the Ui class responsible for splitting the text
into shorter lines as needed, so that it is easier to change the maximum
line length in the future.

chloelee767 committed:
Add generic file reading and writing functions

chloelee767 committed:
Create TaskList class to replace List<Task>

chloelee767 committed:
Remove unnecessary import

chloelee767 committed:
Fix runtest.sh to work with packages

chloelee767 committed:
Update text ui testing for f5dc92cbf420bbff28406efbbed8d8098be739e4

chloelee767 committed:
Complete level 7

chloelee767 committed:
Make Deadlines and Events use dates

chloelee767 committed:
Simplify public methods for TaskListStorage

chloelee767 committed:
Merge branch 'branch-Level-7' into master

chloelee767 committed:
Merge branch 'branch-Level-8' into master

chloelee767 committed:
Fix typo in invalid date message

chloelee767 committed:
Remove BotClass, make Duke implement Bot

chloelee767 committed:
Clean up some of duke's messages/responses

chloelee767 committed:
Make DukeParsingException package-private

Since it's only used in the parser package

chloelee767 committed:
Add jenv file

chloelee767 committed:
Document all public classes and methods plus some others

chloelee767 committed:
Modify code to meet coding standard

chloelee767 committed:
Implement find

chloelee767 committed:
Merge branch 'branch-A-JavaDoc' into master

chloelee767 committed:
Merge branch 'branch-A-CodingStandard' into master

chloelee767 committed:
Merge branch 'branch-Level-9' into master

chloelee767 committed:
Extract repetitive code in command package into functions

chloelee767 committed:
Merge branch 'master' into add-gradle-support

chloelee767 committed:
Make project successfully run with gradle

chloelee767 committed:
Merge branch 'add-gradle-support' into master

khor-jingqian commented on src/main/java/duke/Bot.java:
```diff
@@ -0,0 +1,19 @@
+    /**
+     * Make Duke say the given String.
+     *
+     * @param string the String for Duke to say
```
Should there be punctuation behind the parameter description?

khor-jingqian commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,162 @@
+package duke.parser;
+
+import duke.command.*;
```
Would it be better to list out all the imports for your command package? This might be an issue on Intellij's side as they group together imports when there are several

khor-jingqian commented on src/main/java/duke/storage/FileReadingException.java:
```diff
@@ -0,0 +1,7 @@
+
+class FileReadingException extends Exception {
+    FileReadingException(Exception e) {
+        // TODO
```
Should this TODO be empty? I think you may have missed this after completing the work

khor-jingqian commented on src/main/java/duke/storage/TaskListStorage.java:
```diff
@@ -0,0 +1,144 @@
+    }
+
+    /**
+     * Try to load an existing TaskList from a file (specified in the constructor). A new TaskList is created
```
Should you use "Tries" instead of "Try" to conform to coding standards?

khor-jingqian commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,51 @@
+    /**
+     * Returns the Date when this Deadline is due.
+     *
+     * @return the due Datek
```
Is this a typo?

khor-jingqian commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,64 @@
+    /**
+     * Returns the end date of this Event.
+     *
+     * @return the end ate
```
Is this a typo?

khor-jingqian commented on src/test/java/duke/ui/UiTest.java:
```diff
@@ -0,0 +1,44 @@
+    }
+
+    @Test
+    public void newSay() {
```
Is the implementation meant to be as such?

khor-jingqian reviewed (commented):
Overall, I do not see any issues at all! Really good job. Only some slight nitpicks.

chloelee767 committed:
Fix code style based on PR comments

chloelee767 committed:
Remove dummy JUnit test

chloelee767 committed:
Setup checkstyle and Eclipse formatter rules

chloelee767 commented:
Okay noted, thanks for the feedback!

chloelee767 committed:
Minimal functioning GUI

chloelee767 committed:
Merge branch 'branch-Level-10' into master

chloelee767 committed:
Remove unused unit test and TODOs

lixupengg commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,168 @@
+     */
+    public static Command parse(String input) {
+        input = input.strip();
+        String[] split = input.split("\\s+", 2); // guranteed to contain at least ""
```
Maybe a more representative name could be used to represent the splited string Array? eg splitStringParts

lixupengg commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,168 @@
+                    + "the format event <description> /at <start>-<end>!");
+        }
+
+        Date start = parseDate(dateStrings[0]);
```
Perhaps a more specific name such as startDate instead?

lixupengg commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,168 @@
+     * not exist, or if either of the tokens are blank.
+     */
+    private static String[] splitAround(String string, String pattern) throws DukeParsingException {
+        int n = 2;
```
Maybe a comment should be used to explain what n represents, and why 2 is used (magic number)

lixupengg reviewed (commented):
Good job in general! But with some minor issues as below.

chloelee767 committed:
Rename Bot to Ui

+ Simplify say function in MainWindow
+ Change all commands to call Ui.say only once

chloelee767 committed:
Fix index out of bound exception in DoneCommand

chloelee767 committed:
Rewrite Duke and Gui to be more OOP

chloelee767 committed:
Add a few assertions

chloelee767 committed:
Enable assertions in gradle run

chloelee767 committed:
Improve code quality as per textbook

chloelee767 committed:
Merge pull request #1 from chloelee767/branch-A-Assertions

Complete A-Assertions

chloelee767 committed:
Merge branch 'master' into branch-A-CodeQuality

chloelee767 committed:
Merge pull request #2 from chloelee767/branch-A-CodeQuality

Complete A-CodeQuality

chloelee767 committed:
Add basic task editing functionality

chloelee767 committed:
Fix unsafe generics in EditCommand, extract Edits to package

chloelee767 committed:
Improve code quality

- Document more classes
- Add more unit tests
- General code quality improvements, such as SLAP, extracting methods

chloelee767 committed:
Improve GUI

- Format errors differently
- Improve overall appearance

chloelee767 committed:
Enable xlint unchecked when compiling

chloelee767 committed:
Handle loading empty task list better

Don't throw an error when loading empty task list

chloelee767 committed:
Change which messages are displayed as errors

chloelee767 committed:
Fix clipped text in GUI

chloelee767 committed:
Update user guide

chloelee767 committed:
Set theme jekyll-theme-midnight

chloelee767 committed:
Add screenshot

chloelee767 committed:
Minor improvements to some error messages

chloelee767 committed:
Fix typo in user guide

chloelee767 committed:
Fix user guide not displaying

chloelee767 committed:
Add back support for CLI

chloelee767 committed:
Format JavaDocs
